src/linear_algebra/vandermonde.lean

```diff
@@ -133,4 +134,63 @@ lemma det_vandermonde_ne_zero_iff [is_domain R] {n : ℕ} {v : fin n → R} :
   det (vandermonde v) ≠ 0 ↔ function.injective v :=
 by simpa only [det_vandermonde_eq_zero_iff, ne.def, not_exists, not_and, not_not]
 
+theorem vandermonde_invertibility' {R : Type*} [comm_ring R]
+[is_domain R] {n : ℕ} (v : fin n ↪ R) {f : fin n → R}
+(h₂ : ∀ j, ∑ i : fin n, (v j ^ (i : ℕ)) * f i = 0) : f = 0
+:= eq_zero_of_mul_vec_eq_zero (det_vandermonde_ne_zero_iff.mpr v.inj') (funext h₂)
+
+theorem vandermonde_invertibility {R : Type*} [comm_ring R]
+[is_domain R] {n : ℕ} (v : fin n ↪ R) {f : fin n → R}
+(h₂ : ∀ j, ∑ i, f i * (v j ^ (i : ℕ)) = 0) : f = 0
+:= by {refine vandermonde_invertibility' v _, simp_rw mul_comm, exact h₂}
+
+theorem vandermonde_invertibility_transposed {R : Type*} [comm_ring R]
+[is_domain R] {n : ℕ} (v : fin n ↪ R) {f : fin n → R}
+(h₂ : ∀ i : fin n, ∑ j : fin n, f j * (v j ^ (i : ℕ)) = 0) : f = 0
+:= eq_zero_of_vec_mul_eq_zero (det_vandermonde_ne_zero_iff.mpr v.inj') (funext h₂)
+
 end matrix
+
+namespace polynomial
+open_locale polynomial
+
+theorem eq_zero_iff_degree_lt_equiv_eq_zero {R : Type*} [comm_ring R] {n : ℕ}
+  {p : R[X]} (hp : p ∈ degree_lt R n) : p = 0 ↔ degree_lt_equiv _ _ ⟨p, hp⟩ = 0 :=
```

Comment:
The other direction might make a good `simp` lemma.

Reply:
Don't disagree. I actually had a bunch more of these lemmas and defined a notation for `degree_lt_equiv _ _` but it seemed overkill in the end.